languages/Assembler.groovy

```diff
@@ -340,14 +343,45 @@ def createLinkEditCommand(String buildFile, LogicalFile logicalFile, String memb
 	
 	// define the MVSExec command to link edit the program
 	MVSExec linkedit = new MVSExec().file(buildFile).pgm(props.assembler_linkEditor).parm(parameters)
-
+	
 	// add DD statements to the linkedit command
 	String assembler_loadPDS = props.getFileProperty('assembler_loadPDS', buildFile)
 	String deployType = buildUtils.getDeployType("assembler", buildFile, logicalFile)
 	linkedit.dd(new DDStatement().name("SYSLMOD").dsn("${assembler_loadPDS}($member)").options('shr').output(true).deployType(deployType))
+	
 	linkedit.dd(new DDStatement().name("SYSPRINT").options(props.assembler_tempOptions))
 	linkedit.dd(new DDStatement().name("SYSUT1").options(props.assembler_tempOptions))
 
+	// Create linkEditInstream
+	String sysin_linkEditInstream = ''
+	// linkEdit stream specified
+	if (linkEditStream) {
+		sysin_linkEditInstream += "  " + linkEditStream.replace("\\n","\n").replace('@{member}',member)
+	} else { // dynamically add any required statements via SYSIN
+
+		// add mq stub according to file flags
+		if(buildUtils.isMQ(logicalFile)) {
+			// include mq stub program
+			// https://www.ibm.com/docs/en/ibm-mq/9.3?topic=files-mq-zos-stub-programs
+			sysin_linkEditInstream += buildUtils.getMqStubInstruction(logicalFile)
+		}
+	}
+
+	// Define SYSIN dd
+	if (sysin_linkEditInstream) {
+		if (props.verbose) println("** Generated linkcard input stream: \n $sysin_linkEditInstream")
+		 linkedit.dd(new DDStatement().name("SYSIN").instreamData(sysin_linkEditInstream))
+	}
+
+	// Overwriting SYSLIN with SYSPIN, see
+	// dynalloc macro -- ddname list
+	// https://www.ibm.com/docs/en/zos/2.4.0?topic=facilities-invoking-binder-program-from-another-program
+	//
+	linkedit.dd(new DDStatement().name("SYSPIN").dsn("&&TEMPOBJ").options("SHR"))
+	if (sysin_linkEditInstream) linkedit.dd(new DDStatement().ddref("SYSIN"))
+	linkedit.setDdnames("SYSPIN,,SYSLMOD,SYSLIB,,SYSPRINT,,,,,,,,,,,,,")
```

Comment:
Feedback from reviewers
*  avoid using `DDNames`, which increases complexity and maintainability. This leads to moving away from temporary dataset allocations for the object deck.